### PR TITLE
AYS-34 | Remove the Detailed Phone Number Validation from Filtering Requests

### PR DESCRIPTION
**Pull request by AhmetAksunger**
I've removed the detailed phone number validation from **filtering** requests because it caused some unsolvable bugs.
Instead, there's much simpler validation for the `countryCode` and the `lineNumber` fields.

- `countryCode` : must be between 1 and 3 characters
- `lineNumber` : must be between 7 and 15 characters

Note: Just to make it clear, the validation mentioned above is only used for **filtering** requests.

### Checklist

Before submitting your pull request, ensure the following:

- [x] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](CONTRIBUTING.md#branch-naming-conventions).

- [x] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [x] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [x] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes
      made in Postman.

- [x] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [x] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [x] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

- [x] **Jira Issue Transition:**
    - If this pull request is related to a Jira issue, ensure that the issue is moved to the "In Review" status on the
      Jira board.

### CODE REVIEW

File: src/main/java/com/ays/common/model/dto/request/AysPhoneNumberFilterRequest.java [hunk 1]
```diff
@@ -11,24 +10,25 @@
...
 @Getter
 @Setter
-@PhoneNumber
-public class AysPhoneNumberFilterRequest implements AysPhoneNumberAccessor {
+public class AysPhoneNumberFilterRequest {
 
     /**
      * The country code of the phone number
      */
+    @Size(min = 1, max = 3, message = "COUNTRY CODE MUST BE BETWEEN 1 AND 3 CHARACTERS")
```
Comment (agitrubard) [inline src/main/java/com/ays/common/model/dto/request/AysPhoneNumberFilterRequest.java:18]: Buraya mesaj yazmamıza ihtiyaç yok, sadece minimum ve maximum değerleri belirtmemiz yeterli

File: src/main/java/com/ays/common/model/dto/request/AysPhoneNumberFilterRequest.java [hunk 2]
```diff
@@ -11,24 +10,25 @@
...
     private String countryCode;
 
     /**
      * The line number of the phone number
      */
+    @Size(min = 7, max = 15, message = "LINE NUMBER MUST BE BETWEEN 7 AND 15 CHARACTERS")
```
Comment (agitrubard) [inline src/main/java/com/ays/common/model/dto/request/AysPhoneNumberFilterRequest.java:24]: Buraya mesaj yazmamıza ihtiyaç yok, sadece minimum ve maximum değerleri belirtmemiz yeterli

File: src/main/java/com/ays/common/util/AysJsonUtil.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import com.google.gson.GsonBuilder;
+import lombok.experimental.UtilityClass;
+
+/**
+ * This is a utility class to handle JSON operations within the project.
+ */
+@UtilityClass
+public class AysJsonUtil {
+
+    public static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
```
Comment (agitrubard) [inline src/main/java/com/ays/common/util/AysJsonUtil.java:13]: `GSON` objesini `private` yapabiliriz

**Review by agitrubard — CHANGES_REQUESTED** [review]
Eline sağlık genel olarak okey, 1-2 yorumum oldu o kısmı hızlıca düzeltirsek sonrasında mergeleyebiliriz

File: src/main/java/com/ays/common/model/dto/request/AysPhoneNumberFilterRequest.java (hunk 1 above)
Comment (AhmetAksunger) [reply src/main/java/com/ays/common/model/dto/request/AysPhoneNumberFilterRequest.java]: Kaldırdım

File: src/main/java/com/ays/common/util/AysJsonUtil.java (hunk 3 above)
Comment (AhmetAksunger) [reply src/main/java/com/ays/common/util/AysJsonUtil.java]: Değiştirdim

**Review by agitrubard — APPROVED** [review]

**Review by ibrahimdenizz — APPROVED** [review]

**Pull request merged**